pkg/corerp/backend/controller/createorupdateresource.go

```diff
@@ -60,16 +65,66 @@ func getDataModel(id resources.ID) (v1.DataModelInterface, error) {
 	}
 }
 
+func isPortableResource(resourceTypeResourceObj *database.Object) (bool, error) {
+	data := resourceTypeResourceObj.Data
+	if data == nil {
+		return false, errors.New("resource type's data is nil. cannot determine if resource is portable")
+	}
+
+	// Mostly this condition should not be hit as the resource type should have properties.
+	properties, ok := data.(map[string]interface{})["properties"]
+	if !ok {
+		return false, errors.New("resource type's properties not found. cannot determine if resource is portable")
+	}
+
+	capabilities, ok := properties.(map[string]interface{})["capabilities"]
+	if !ok {
+		// Could be a resource type that does not have capabilities field. In that case, it is not a portable resource.
+		// Ex: application, environment, radius resource group
+		return false, nil
+	}
+
+	for _, capability := range capabilities.([]interface{}) {
+		if capability == isPortable {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
 // Run checks if the resource exists, renders the resource, deploys the resource, applies the
 // deployment output to the resource, deletes any resources that are no longer needed, and saves the resource.
 func (c *CreateOrUpdateResource) Run(ctx context.Context, request *ctrl.Request) (ctrl.Result, error) {
+	// This code is general and we might be processing an async job for a resource or a scope, so using the general Parse function.
```

Comment:
I think there is room for abstraction/refactoring both in this controller and deploymentprocessor. The code is getting too verbose/difficult to read.